sid.py

```diff
@@ -299,12 +326,14 @@ def validate_items(self, items):
         for item in items:
             for key in item:
                 if key == 'type':
-                    if type(item[key]) != str or not re.match(r'Module$|Submodule$|feature$|identity$|node$|notification$|rpc$|action$', item[key]):
+                    matched = re.match(r'Module$|Submodule$|feature$|identity$|node$|notification$|rpc$|action$', item[key])
+                    nstrtype = (type(item[key]) != unicode)
+                    if nstrtype or matched is None:
                         raise SidFileError("invalid 'type' value '%s'." % item[key])
                     continue
 
                 if key == 'label':
-                    if type(item[key]) != str:
+                    if type(item[key]) != unicode:
```

Comment:
unicode doesn't work with python3 I guess you need to prefer use `try/except NameError` or something like that:
```python
try:
    # python2 first
    # fail with python 3
    check = type(item[key]) != unicode
except NameError:
    check = type(item[key]) != str
```
If you decide that we need to support python2 and 3 then like your changes on imports you also need to keep backward/forward here and at line 339.

- https://stackoverflow.com/questions/4987327/how-do-i-check-if-a-string-is-unicode-or-ascii
- https://docs.python.org/3/whatsnew/3.0.html